app/views/sites/default.html.erb

```diff
@@ -1,10 +1,21 @@
-<%= render(
-  HeroImageComponent.new(
-    image_path: @site.hero_image,
-    image_credit: @site.hero_image_credit
+<% if @site.hero_text.present? %>
+  <%= render(
+    HeroImageComponent.new(
+      image_path: @site.hero_image,
+      image_credit: @site.hero_image_credit,
+      title: @site.hero_text
+      )
     )
-  )
-%>
+  %>
+<% else %>
+  <%= render(
+    HeroImageComponent.new(
+      image_path: @site.hero_image,
+      image_credit: @site.hero_image_credit
+      )
+    )
+  %>
+<% end %>
```

Comment:
Instead of using an if/else to render this, could you could just pass in @site.hero_text and handle the logic for whether to show it or not in the hero image component? 

Instead of setting the title as a default in the params there, you could just do something like
`@title = title.presence || 'default title'` in the init

Reply:
I tried to wrap just the  `title` argument bit in an if/else but it kept failing to run. This looks loads better, cheers.